packages/components/src/sandbox/index.js

```diff
@@ -215,23 +221,21 @@ export default function Sandbox( {
 		ref.current.addEventListener( 'load', tryNoForceSandbox, false );
 		defaultView.addEventListener( 'message', checkMessageForResize );
 
+		const iframeRef = ref.current;
+
 		return () => {
-			ref.current?.removeEventListener(
-				'load',
-				tryNoForceSandbox,
-				false
-			);
+			iframeRef?.removeEventListener( 'load', tryNoForceSandbox, false );
 			defaultView.addEventListener( 'message', checkMessageForResize );
 		};
-	}, [] );
+	}, [ trySandbox ] );
```

Comment:
This `useEffect` was supposed to run only when the component mounts, but with the changes from this PR it will run every time `trySandbox` changes — i.e any time any of its dependencies change.